lib/components/narrative/narrative-itineraries-header.tsx

```diff
@@ -172,48 +202,26 @@ export default function NarrativeItinerariesHeader({
                 )}
               </StyledIconWrapper>
             </button>
-            <select
-              aria-label={intl.formatMessage({
-                id: 'components.NarrativeItinerariesHeader.sortBy'
-              })}
-              onBlur={onSortChange}
-              onChange={onSortChange}
-              title={intl.formatMessage({
-                id: 'components.NarrativeItinerariesHeader.sortBy'
-              })}
-              value={sort.type}
+            <SortResultsDropdown
+              id="sort-results"
+              label={sortResultsLabel}
+              name={sortText}
+              title={sortResultsLabel}
             >
-              <option value="BEST">
-                {intl.formatMessage({
-                  id: 'components.NarrativeItinerariesHeader.selectBest'
-                })}
-              </option>
-              <option value="DURATION">
-                {intl.formatMessage({
-                  id: 'components.NarrativeItinerariesHeader.selectDuration'
-                })}
-              </option>
-              <option value="ARRIVALTIME">
-                {intl.formatMessage({
-                  id: 'components.NarrativeItinerariesHeader.selectArrivalTime'
-                })}
-              </option>
-              <option value="DEPARTURETIME">
-                {intl.formatMessage({
-                  id: 'components.NarrativeItinerariesHeader.selectDepartureTime'
-                })}
-              </option>
-              <option value="WALKTIME">
-                {intl.formatMessage({
-                  id: 'components.NarrativeItinerariesHeader.selectWalkTime'
-                })}
-              </option>
-              <option value="COST">
-                {intl.formatMessage({
-                  id: 'components.NarrativeItinerariesHeader.selectCost'
-                })}
-              </option>
-            </select>
+              {sortOptionsArr.map((x) => {
+                return (
```

Comment:
Nitpick: this can probably be inlined

Reply:
[Conflicting reviews on this one](https://github.com/opentripplanner/otp-react-redux/pull/897#discussion_r1198265068)

Reply:
Go with Binh's! That's clean

Reply:
I just meant removing the `return`!